library_management/models/library_book.py

```diff
@@ -0,0 +1,162 @@
+from markupsafe import Markup
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+import re
+
+class LibraryManagement(models.Model):
+    _name = 'library.book'
+    _description = 'Book in the library'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _rec_name = 'title'
+
+    title = fields.Char(string="Title", required=True)
+    isbn = fields.Char(string="ISBN", size=17, help="13-Digits ISBN number")
+    publication_date = fields.Date(string="Publication Date")
+    author_id = fields.Many2one('library.author', string="Author")
+    image_1920 = fields.Binary(string="Cover image")
+    book_age = fields.Integer(string="Book Age (Years)", compute="_compute_book_age", store=True)
+    member_id = fields.Many2one('library.member',tracking=True, string="Borrowing by")
+
+
+    rental_fee = fields.Monetary(string="Rental Fee", default=1.0)
+    currency_id = fields.Many2one(
+        'res.currency',
+        string="Currency",
+        compute='_compute_currency',
+        store=True,
+        readonly=True,
+        required=True
+    )
+
+
+
+    total_rental = fields.Float(string="Total Rental", compute="_compute_total_rental", store=True)
+    status = fields.Selection([
+        ('available', 'Available'),
+        ('borrowed', 'Borrowed'),
+        ('lost', 'Lost')
+    ], string="Status", default='available',
+        tracking=True,
+        required=True, group_expand="_read_group_stage_ids")
+    genre = fields.Selection([
+        ('fiction', 'Fiction'),
+        ('nonfiction', 'Non-Fiction'),
+        ('fantasy', 'Fantasy'),
+        ('biography', 'Biography'),
+        ('science', 'Science'),
+    ], string="Genre")
+
+    @api.depends('member_id', 'message_ids')
+    def _compute_total_rental(self):
+        pattern = re.compile(r'Rental fee:\s*\$?([0-9,.]+)')
+        for book in self:
+            total = 0.0
+            for msg in book.message_ids:
+                if msg.body:
+                    # Search for "Rental fee: $xxx.xx" pattern
+                    match = pattern.search(msg.body)
+                    if match:
+                        # Remove commas and convert to float
+                        amount_str = match.group(1).replace(',', '')
+                        try:
+                            amount = float(amount_str)
+                            total += amount
+                        except ValueError:
+                            pass
```

Comment:
```suggestion
                            raise ValidationError(.....)
```
should put raise ValidationError instead of pass